helios-client/src/main/java/com/spotify/helios/common/JobValidator.java

```diff
@@ -212,6 +215,24 @@
     return errors;
   }
 
+  private Set<String> validateJobHostName(final String hostname) {
+    final Set<String> errors = Sets.newHashSet();
+
+    // we're fine if no hostname is set
+    if (hostname == null || hostname.isEmpty()) {
```

Comment:
Nitpick: use `Strings.isNullOrEmpty()` as we do that throughout the rest of the code.